azurelinuxagent/ga/exthandlers.py

```diff
@@ -1099,20 +1102,27 @@ def collect_ext_status(self, ext):
         ext_status = ExtensionStatus(seq_no=seq_no)
         try:
             data_str = fileutil.read_file(ext_status_file)
+
+            if len(data_str) > MAX_STATUS_FILE_SIZE_IN_BYTES:
+                msg = "Handler - {0}, status file {1} of size {2} bytes is too big. Max Limit allowed is {3} " \
+                      "bytes".format(ext.name, ext_status_file, len(data_str), MAX_STATUS_FILE_SIZE_IN_BYTES)
+                logger.periodic_warn(logger.EVERY_DAY, msg)
+                raise IOError(msg)
+
             data = json.loads(data_str)
             parse_ext_status(ext_status, data)
         except IOError as e:
-            ext_status.message = u"Failed to get status file {0}".format(e)
+            ext_status.message = u"Failed to get status file: {0}".format(e)
             ext_status.code = -1
-            ext_status.status = "error"
+            ext_status.status = EXTENSION_STATUS_ERROR
```

Comment:
I believe these were going to change to warning?

Reply:
Yes. Will sync with the Windows side and check how to make it work. They don't send `ext_status.code = -1` and only send `ext_status.code = ExtensionErrorCodes.PluginSettingsStatusInvalid`. I am currently seeing where `-1` comes into play.